pkg/apply/solver/solver.go

```diff
@@ -86,127 +97,148 @@ type resourceObjects interface {
 	IdsForPrevInv() []object.ObjMetadata
 }
 
-// BuildTaskQueue takes a set of resources in the form of info objects
-// and constructs the task queue. The options parameter allows
-// customization of how the task queue are built.
-func (t *TaskQueueSolver) BuildTaskQueue(ro resourceObjects,
-	o Options) *TaskQueue {
-	var invAddCounter int
-	var invReplaceCounter int
-	var applyCounter int
-	var pruneCounter int
-	var waitCounter int
+// Build returns the queue of tasks that have been created.
+func (t *TaskQueueBuilder) Build() *TaskQueue {
+	return &TaskQueue{
+		tasks: t.tasks,
+	}
+}
+
+// AppendInvAddTask appends an inventory add task to the task queue.
+// Returns a pointer to the Builder to chain function calls.
+func (t *TaskQueueBuilder) AppendInvAddTask(ro resourceObjects) *TaskQueueBuilder {
```

Comment:
Do we need to expose this to callers? It seemed like this is something that can be handled inside the `Solver/TaskQueueBuilder`.

Reply:
We can add a layer on top of the builder, but I think this is a reasonable step to build the task queue in a more dynamic way. I think we can firm up the API when we get to "depends-on".

Reply:
Agree